### ISSUE TITLE
Feature Interaction Extension

### ISSUE THREAD

**Issue by bimhud** [pycaret/pycaret#1338]
**Is your feature request related to a problem? Please describe.**
Feature interaction is really useful to discover the new combination and help for business explanation.
The current implementation should enable to have a choice to select which features to be interacted with as the business might have indicators for which features could be useful for interaction. This helps to reduce the memory issue when trying not to combine all possible features. 

**Describe the solution you'd like**
Provide a list of features up front from the ``setup`` if the user wanted to see the interaction within those features.

**Describe alternatives you've considered**
We have to do feature interaction upfront before starting the ``setup``. However, this won't be the best solution as the  interaction might coming  from the encoded categorical features as well.
I would like to see this as a part of the pipeline.
We can use a custom pipeline, however, it slowing down the speed of development and not utilize what has been done from Pycaret.

**Comment by moezali1** [comment]
Closing as part of the 4.0 backlog cleanup — see #4177.

PyCaret 4.0 is a ground-up rewrite (OOP-only API, sklearn-native engine, Python 3.11+, sklearn ≥ 1.7) and almost all pre-4.0 issues either reference removed APIs (`setup()`, `plot_model`, `evaluate_model`, `interpret_model`, `automl`, the legacy time-series module, the Streamlit dashboard, …) or no longer reproduce on the new code path.

**If this still affects you in 4.0, please re-file as a new issue** with:

- the experiment constructor call (`Experiment(...)`),
- the operation that fails,
- the full traceback,
- your Python / sklearn / pycaret versions.

Resources for 4.0:

- Docs: https://pycaret.org/docs
- Migrate from 3.x: https://pycaret.org/docs/guides/migrating-from-3-x
- API reference: https://pycaret.org/reference
- FAQ: https://pycaret.org/docs/resources/faqs

Thanks for filing — and apologies for the long latency. The repo had drifted; we're starting clean.
